src/components/ConfirmationDialog.js

```diff
@@ -74,8 +74,8 @@ export default class ConfirmationDialog extends React.Component<ConfirmationDial
 
   render() {
     const okButtonStyle = this.props.dangerous ? 'danger' : 'primary';
-    return this.props.show ? (
-      <Modal show>
+    return (
+      <Modal show={this.props.show} onHide={this.props.onCancel}>
```

Comment:
Why was this `onHide` hook added? The `onCancel` prop is already passed into the `onClick` hook of the cancel button on line 86. Will this cause the `onCancel` callback to execute twice?

Reply:
I updated this to use the built in show functionality and also eliminated some console warnings I saw. This won't call the `onCancel` function twice. The `onHide` function is only called if you click a non-static backdrop or the "X" button. The Cancel button does not trigger onHide, it's just a button that calls whatever function you pass to it. In this case, that function also happens to be `onCancel`. https://react-bootstrap.github.io/components/modal/#modals-live